### ISSUE TITLE
Crashing Server

### ISSUE THREAD

**Issue by CalendulaX** [Chikorita-Lover/Caffeinated#9]
this mod oddly keeps crashing my server which is fabric 1.20.1. The Crash Report is uploaded. Please check the reason and fix it maybe?
[crash-2023-12-28_16.30.21-server.txt](https://github.com/Chikorita-Lover/Caffeinated/files/13784747/crash-2023-12-28_16.30.21-server.txt)

**Comment by Chikorita-Lover** [comment]
When does the crash occur? Is it when a specific portion of the world is loaded, or does it just occur randomly?

**Comment by CalendulaX** [comment]
I believe it’s crashing when unloaded chunks loading.

Chikorita Lover ***@***.***>于2024年1月7日 周日2:56写道：

**Comment by stunlocked1** [comment]
> When does the crash occur? Is it when a specific portion of the world is loaded, or does it just occur randomly?

I have the same issue it says
```
java.lang.IllegalArgumentException: Cannot get property EnumProperty{name=half, clazz=class net.minecraft.block.enums.DoubleBlockHalf, values=[upper, lower]} as it does not exist in Block{minecraft:air}
	at MC//net.minecraft.state.State.get(State.java:98)
	at com.chikoritalover.caffeinated.block.FloweringCoffeeShrubBlock.randomTick(FloweringCoffeeShrubBlock.java:50)
```
seems to happen randomly when just exploring the world

https://gist.github.com/stunlocked1/ea6938b3fab254aefc75b880ce0ad3aa

**Comment by Chikorita-Lover** [comment]
Okay, I think I may know what the issue is. I'll post a hotfix here in a day or two. Thanks for the report!

**Comment by Chikorita-Lover** [comment]
Could either of you send one of the seeds where this crash occurs? The crash reports provide the position but not the world seed itself.

**Comment by Neil2250** [comment]
[crash-2024-05-03_00.03.02-server.txt](https://github.com/Chikorita-Lover/Caffeinated/files/15194889/crash-2024-05-03_00.03.02-server.txt)

Also had a crash on SP, can't repeat the conditions, though. Was going to do a new post but it appears similar enough. I believe it's related to your flowering coffee shrub, but i've gone several worlds with jungles before experiencing this problem, so i'm at a loss as to what could be causing it exactly.

I can't send seed/etc as i'm using a heavily edited world generator, but the only thing I could think could be that the top of the block causes a fatal crash if a block update causes the bottom to pop off without the top realising it (i.e., during terrain placement, if you have a mod that features something that replaces certain types of grass, like the immersive weathing mod's grass silt, or loam, or Blooming Nature's Laterlit). 

I have very little experience diagnosing this kind of thing, but it could be also related to the storing of data given it happens on exploration, so it may have something to do with chunk "sections" with vertical chunks. https://wiki.vg/Chunk_Format#:~:text=However%2C%20these%20are%20not%20the,(chunk%20sections%20are%20cubic).

Hope these can be leads to follow. I enjoy this mod. Keep it up! xx

**Comment by emihead** [comment]
can you backport this fix to 1.20.1? i just had this crash
